relayer/relayer/src/redis/mod.rs

```diff
@@ -0,0 +1,165 @@
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc, Mutex,
+};
+
+use log::{debug, info, warn};
+use redis::{Client as RedisClient, Commands, Connection, RedisError};
+use thiserror::Error;
+use tokio::{
+    select,
+    sync::{broadcast, mpsc},
+};
+
+use crate::{config::Config, CircuitBreakerEvent};
+
+pub const ETH_BLOCK_KEY: &str = "ethereum_block_number";
+pub const ALEPH_BLOCK_KEY: &str = "alephzero_block_number";
+
+#[derive(Debug, Error)]
+#[error(transparent)]
+#[non_exhaustive]
+pub enum RedisManagerError {
+    #[error("redis connection error")]
+    Redis(#[from] RedisError),
+
+    #[error("channel send error")]
+    Send(#[from] mpsc::error::SendError<u32>),
+
+    #[error("channel broadcast error")]
+    Broadcast(#[from] broadcast::error::SendError<u32>),
+
+    #[error("channel receive error")]
+    Receive(#[from] broadcast::error::RecvError),
+}
+
+pub struct RedisManager;
+
+impl RedisManager {
+    pub async fn run(
+        is_first_run: Arc<AtomicBool>,
```

Comment:
No sure why you wouldn't pass just a plain `bool` here, it actually should be safer

Reply:
if e.g. other components need to exclusively access it & alter - but I guess we can always change it